### PR TITLE
PLANNER-1301-ColorPublishedTimeslot

### PR DESCRIPTION
**Pull request by MusaTalluzi**
Color scheduled cells if they're different than published ones

### CODE REVIEW

File: optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingXlsxFileIO.java [hunk 1]
```diff
@@ -1013,10 +1013,14 @@ private void writeTalkList() {
                 nextCell().setCellValue(talk.getFavoriteCount());
                 nextCell().setCellValue(talk.getCrowdControlRisk());
                 nextCell(talk.isPinnedByUser() ? pinnedStyle : defaultStyle).setCellValue(talk.isPinnedByUser());
-                nextCell().setCellValue(talk.getTimeslot() == null ? "" : DAY_FORMATTER.format(talk.getTimeslot().getDate()));
-                nextCell().setCellValue(talk.getTimeslot() == null ? "" : TIME_FORMATTER.format(talk.getTimeslot().getStartDateTime()));
-                nextCell().setCellValue(talk.getTimeslot() == null ? "" : TIME_FORMATTER.format(talk.getTimeslot().getEndDateTime()));
-                nextCell().setCellValue(talk.getRoom() == null ? "" : talk.getRoom().getName());
+                nextCell(talk.getTimeslot().equals(talk.getPublishedTimeslot()) ? defaultStyle : hardPenaltyStyle)
```
Comment (ge0ffrey) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingXlsxFileIO.java]: hardPenaltyStyle? The timeslot one is a medium constraint and the room one is a soft constraint.

Let's use a different color (= style) altogether, maybe something like republishedStyle? Not green (= good), not yellow/orange/red (used by hard soft) not purple (used by pinned). So how about blue or pink?

Comment (MusaTalluzi) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingXlsxFileIO.java]: Blue implies something is ok (it's close to green), so I think pink is better (or magenta maybe?)

Comment (ge0ffrey) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingXlsxFileIO.java]: Sounds good

File: optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingXlsxFileIO.java [hunk 2]
```diff
@@ -1013,10 +1013,14 @@ private void writeTalkList() {
                 nextCell().setCellValue(talk.getFavoriteCount());
                 nextCell().setCellValue(talk.getCrowdControlRisk());
                 nextCell(talk.isPinnedByUser() ? pinnedStyle : defaultStyle).setCellValue(talk.isPinnedByUser());
-                nextCell().setCellValue(talk.getTimeslot() == null ? "" : DAY_FORMATTER.format(talk.getTimeslot().getDate()));
-                nextCell().setCellValue(talk.getTimeslot() == null ? "" : TIME_FORMATTER.format(talk.getTimeslot().getStartDateTime()));
-                nextCell().setCellValue(talk.getTimeslot() == null ? "" : TIME_FORMATTER.format(talk.getTimeslot().getEndDateTime()));
-                nextCell().setCellValue(talk.getRoom() == null ? "" : talk.getRoom().getName());
+                nextCell(talk.getTimeslot().equals(talk.getPublishedTimeslot()) ? defaultStyle : republishedStyle)
```
Comment (ge0ffrey) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingXlsxFileIO.java]: @MusaTalluzi This caused an NPE when running the Generator ecause getTimeslot() returns null.
I've fixed it my branch by using java.util.Objects.equal() instead.

**Pull request merged**